framework/src/transfers/MultiAppInterpolationTransfer.C

```diff
@@ -52,25 +48,28 @@ validParams<MultiAppInterpolationTransfer>()
 }
 
 MultiAppInterpolationTransfer::MultiAppInterpolationTransfer(const InputParameters & parameters)
-  : MultiAppTransfer(parameters),
-    _to_var_name(getParam<AuxVariableName>("variable")),
-    _from_var_name(getParam<VariableName>("source_variable")),
+  : MultiAppFieldTransferInterface(parameters),
     _num_points(getParam<unsigned int>("num_points")),
     _power(getParam<Real>("power")),
     _interp_type(getParam<MooseEnum>("interp_type")),
     _radius(getParam<Real>("radius"))
 {
   // This transfer does not work with DistributedMesh
   _fe_problem.mesh().errorIfDistributedMesh("MultiAppInterpolationTransfer");
+
+  mooseAssert(_to_var_name.size() == 1 && _from_var_name.size() == 1,
+              " Support single variable only ");
 }
 
 void
 MultiAppInterpolationTransfer::initialSetup()
 {
+  MultiAppFieldTransferInterface::initialSetup();
+
   if (_direction == TO_MULTIAPP)
-    variableIntegrityCheck(_to_var_name);
+    variableIntegrityCheck(_to_var_name[0]);
```

Comment:
this integrity check looks to me should be in `MultiAppFieldTransferInterface::initialSetup`. Then here we do not need to override the functionl

Reply:
Right